src/components/Avatar.js

```diff
@@ -41,14 +47,48 @@ const defaultProps = {
     fallbackIcon: Expensicons.FallbackAvatar,
 };
 
-class Avatar extends PureComponent {
+class Avatar extends Component {
     constructor(props) {
         super(props);
         this.state = {
             imageError: false,
         };
     }
 
+    shouldComponentUpdate(nextProps, nextState) {
+        if (
+            lodashGet(this.props.network, 'isOffline')
+                  && !lodashGet(nextProps.network, 'isOffline')
+        ) {
+            return true;
+        }
+        if (this.props.source !== nextProps.source) {
+            return true;
+        }
+        if (this.state.imageError !== nextState.imageError) {
+            return true;
+        }
+
+        if (this.props.fill !== nextProps.fill) {
+            return true;
+        }
+        if (this.props.size !== nextProps.size) {
+            return true;
+        }
+
+        return false;
+    }
+
+    componentDidUpdate(prevProps) {
+        if (
+            !this.state.imageError || !lodashGet(prevProps.network, 'isOffline')
+            || lodashGet(this.props.network, 'isOffline')
+        ) {
```

Comment:
I think this might be easier to read:
```suggestion
        const isReconnecting = prevProps.network.isOffline && !this.props.network.isOffline;
        if (!this.state.imageError || !isReconnecting) {
```

Reply:
Meanwhile @hungvu193 let's do it.